### PR TITLE
Add modified 'direct' Go Apache Beam runner that uses disksort GBK

### PR DESCRIPTION
**Pull request by schroederc**
Using a disksort for direct CoGBK operations allows the pipeline to
support much larger input sizes without running out of memory while
buffering values locally.

### CODE REVIEW

File: third_party/beam/README.google [hunk 1]
```diff
@@ -0,0 +1,6 @@
+URL: https://github.com/apache/beam
+Version: 72cbd99d6b62bc7ed16dbd1288cd61d54e8bda37
+License: Apache License 2.0: https://opensource.org/licenses/Apache-2.0
+Local Modifications:
+  - Narrow checkout of sdks/go/pkg/beam/runners/direct
+  - Modified package to use disksort for CoGBK implementation
```
Comment (danielmoy-google) [inline third_party/beam/README.google]: do we need to talk at all about the BUILD file or does that not count here?

File: third_party/beam/sdks/go/pkg/beam/runners/direct/gbk.go [hunk 2]
```diff
@@ -57,88 +108,248 @@ func (n *CoGBK) StartBundle(ctx context.Context, id string, data exec.DataManage
...
+	val, err := i.n.valDec[i.idx].Decode(bytes.NewBuffer(iv.Value))
+	if err != nil {
+		return exec.FullValue{}, fmt.Errorf("error decoding value: %v", err)
+	}
+	return val, nil
+}
+
+func (i *iterStream) next() (*iterValue, error) {
+	if i.prev != nil {
+		var prevItems int
```
Comment (danielmoy-google) [inline third_party/beam/sdks/go/pkg/beam/runners/direct/gbk.go]: unused prevItems?

I mean I see you incrementing it, but then... nothing

File: third_party/beam/sdks/go/pkg/beam/runners/direct/gbk.go [hunk 3]
```diff
@@ -18,37 +18,88 @@ package direct
 import (
```
Comment (danielmoy-google) [inline third_party/beam/sdks/go/pkg/beam/runners/direct/gbk.go]: this file with all the changes have a test upstream somewhere?

**Comment by danielmoy-google** [comment]
Looks like it buried two of my comments: unused prevItems int variable, and general question on whether gbk.go has a test or not

File: third_party/beam/sdks/go/pkg/beam/runners/direct/gbk.go (hunk 3 above)
Comment (schroederc) [reply third_party/beam/sdks/go/pkg/beam/runners/direct/gbk.go]: I added some tests in gbk_test.go

File: third_party/beam/sdks/go/pkg/beam/runners/direct/gbk.go (hunk 2 above)
Comment (schroederc) [reply third_party/beam/sdks/go/pkg/beam/runners/direct/gbk.go]: ✓

File: third_party/beam/README.google (hunk 1 above)
Comment (schroederc) [reply third_party/beam/README.google]: `¯\_(ツ)_/¯`

**Pull request merged**